src/sagemaker/session.py

```diff
@@ -170,6 +170,14 @@ def default_bucket(self):
             elif error_code == 'OperationAborted' and 'conflicting conditional operation' in message:
                 # If this bucket is already being concurrently created, we don't need to create it again.
                 pass
+            elif error_code == 'TooManyBuckets':
+                try:
+                    s3.meta.client.head_bucket(Bucket=default_bucket)
+                    LOGGER.info('S3 bucket {} already exists'.format(
```

Comment:
I don't think we need this logging message, in the normal case we don't log this.